stdlib/public/core/Unmanaged.swift

```diff
@@ -34,7 +34,15 @@ public struct Unmanaged<Instance: AnyObject> {
   public static func fromOpaque(
     @_nonEphemeral _ value: UnsafeRawPointer
   ) -> Unmanaged {
-    return Unmanaged(_private: unsafeBitCast(value, to: Instance.self))
+    // NOTE: `value` is allowed to be a dangling pointer, so 
+    // this function must not ever try to dereference it. For
+    // example, it must NOT go through the init(_private:) initializer
+    // because it requires us to materialize a strong reference to 'Instance'.
```

Comment:
```suggestion
    // example, this function must NOT use the init(_private:) initializer
    // because doing so requires materializing a strong reference to 'Instance'.
```